lib/modules/manager/flux/types.ts

```diff
@@ -37,16 +37,28 @@ export interface HelmRepository extends KubernetesResource {
   };
 }
 
-export type FluxResource = HelmRelease | HelmRepository;
+export interface GitRepository extends KubernetesResource {
+  kind: 'GitRepository';
+  spec: {
+    ref: {
+      tag?: string;
+      commit?: string;
+    };
+    url: string;
+  };
+}
+
+export type FluxResource = HelmRelease | HelmRepository | GitRepository;
 
 export interface FluxFile {
   file: string;
 }
 
 export interface ResourceFluxManifest extends FluxFile {
   kind: 'resource';
-  releases: HelmRelease[];
-  repositories: HelmRepository[];
+  helmReleases: HelmRelease[];
```

Comment:
that's not the right way. this interface represent a yaml kubernetes manifest file, so it shouldn't users for other purpose.

Reply:
Hey @viceice, I just learned TypeScript 10 hours ago so execuse me for not knowing.
But the interface was already not representing a single file before my changes.
Knowing that, do you wish me to work on isolating manifest objects and update the code logic accordingly?

Reply:
don't worry. was just a short review. will check the code tomorrow and make more suggestions then

Reply:
Hey @viceice , I think I know why. Because there can be multiple resources in the same yaml file, so that is why it was written that way. So the way I extended the code is okay.
Will mark this as resolved.